src/compiler/checker.ts

```diff
@@ -6609,16 +6608,18 @@ namespace ts {
         }
 
         function getTypeReferenceType(node: TypeReferenceNode | ExpressionWithTypeArguments | JSDocTypeReference, symbol: Symbol) {
+            const typeArguments = typeArgumentsFromTypeReferenceNode(node); // Do unconditionally so we mark type arguments as referenced.

```

Comment:
Found some duplicate code, so we call `typeArgumentsFromTypeReferenceNode` in fewer places now.
(The two parts of duplicate code were even added in the same commit: e223b2e53c)

As for performance, here are my measurements on a sample script:

```
{
    { type T = X<A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P>; }
    { type T = X<A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P>; }
    ... 1000 times ...
}
```

Tested with `node ..\..\TypeScript\built\local\tsc.js --noEmit --diagnostics --noLib`, and shimmed lib with e.g. `interface Array {}` so it would compile.

This PR:

    Memory Used: 30702K 30379K 30134K 30140K 30114K (average: 30293.8K)
    Check Time: 0.11s 0.11s 0.10s 0.10s 0.10s (average: 0.104s)

This PR with 1000 extra calls to `typeArgumentsFromTypeReferenceNode`:

    Memory Used: 37329K 37266K 37830K 37825K 37247K (average: 37499.4K)
    Check Time: 0.85s 0.83s 0.84s 0.84s 0.83s (average: 0.838s)

The memory difference is 7205K, a 0.02% difference per iteration.
The time difference is 0.734s, a 0.7% difference per iteration.